src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
          List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
          Printf.printf "About to fixup the fd\n%!";
-         ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+         Lwt.catch
+          (fun () ->
+            ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+            Lwt.return ()
+          )
+          (fun e ->
+            List.iter (fun fd -> try Unix.close fd with _ -> ()) newfds;
+            Printf.printf "Caught exception: %s\n" (Printexc.to_string e);
+            Lwt.return ()) >>= fun _ ->
          loop ()
```

Comment:
We don't close the server socket `fd_sock` in this function, but I guess that's not a disaster, because it will be open as long as the program is running, and when it terminates it will be cleaned up anyway. (See https://stackoverflow.com/questions/8175827/what-happens-if-i-dont-call-fclose-in-a-c-program#8176030)